### PR TITLE
daemon: fix filtering of service-control changes for snap.app

### PR DESCRIPTION
**Pull request by bboozzoo**
When a user called `snap restart lxd.daemon`, the change would contain
`[lxd.daemon]` in the snap-names. Later, when one called `snap changes lxd`, the
change would not show up.

Fixes: https://bugs.launchpad.net/snapd/+bug/1880560

### CODE REVIEW

File: daemon/api.go [hunk 1]
```diff
@@ -2394,6 +2398,21 @@ func getLogs(c *Command, r *http.Request, user *auth.UserState) Response {
 	}
 }
 
+func namesToSnapNames(inst *servicestate.Instruction) []string {
+	seen := make(map[string]struct{}, len(inst.Names))
```
Comment (zyga) [inline daemon/api.go:2402]: IIRC we usually use `map[string]bool` for this sort of thing.

File: daemon/api.go [hunk 2]
```diff
@@ -2427,7 +2446,8 @@ func postApps(c *Command, r *http.Request, user *auth.UserState) Response {
 	}
 	st.Lock()
 	defer st.Unlock()
-	chg := newChange(st, "service-control", fmt.Sprintf("Running service command"), tss, inst.Names)
+	// names can be snap or snap.app
```
Comment (anonymouse64) [inline daemon/api.go:2449]: maybe mention that now all names will be just the snap name, but in previous versions could have been snap or snap.app?

Comment (bboozzoo) [reply daemon/api.go:2449]: Hmm, but they can still be snap or snap.app. The bug was in storing the name as it was received in `snap-names`, so we ended up with `lxd.daemon` in the list, what was later interpreted as a snap name. Perhaps I should clarify the comment.

Comment (anonymouse64) [reply daemon/api.go:2449]: how can it be snap.app now? It seems that `namesToSnapNames` is implemented such that only the snapName is put into seen, thus only the snapName is put into names and stored through this? Is there some other task/change that is storing as snap.app ?

Comment (bboozzoo) [reply daemon/api.go:2449]: I mean you can still issue `snap restart lxd.daemon`, so inst.Names can still be snap or snap.app.

Comment (anonymouse64) [reply daemon/api.go]: makes sense now, thank you for adjusting the comment

**Pull request merged**